src/arguments/api_test.py

```diff
@@ -37,26 +38,56 @@ def get_request(cls):
 
         print("Trying ...\u26A1")
+        return {
+            "request_url" : request_url,
+            "request_headers" : request_headers,
+        }
 
-        # Make GET request and store the response in response_data.json
-        try:
-            response = requests.get(request_url, headers=request_headers)
-            print(f"Reponse Status Code: {response.status_code}")
-            response_data = json.loads(response.content)
-            parsed_json = json.dumps(response_data, indent=4)
-            output_json = highlight(parsed_json, lexers.JsonLexer(),
-                                    formatters.TerminalFormatter())
-            print(output_json)
-
-            store_data = input('Store response data? (Y/N): ')
-            if(store_data == 'Y' or store_data == 'y'):
+    #saves the json response into a file
+    @classmethod
+    def save_response_data(cls,response_data):
+        store_data = input('Store response data? (Y/N): ')
+        if(store_data == 'Y' or store_data == 'y'):
                 with open('response_data.json', 'w') as jsonFile:
                     json.dump(response_data, jsonFile, indent=4)
                 print("Response data stored in response_data.json")
+    # formats the response data and prints it in json on console
+    @classmethod
+    def print_response_json(cls,response):
+        print(f"Reponse Status Code: {response.status_code}")
+        response_data = json.loads(response.content)
+        parsed_json = json.dumps(response_data, indent=4)
+        output_json = highlight(parsed_json, lexers.JsonLexer(),
+                                    formatters.TerminalFormatter())
+        print(output_json)
+        return response_data
+
+    # Make GET request
+    @classmethod
+    def get_request(cls):
+        request_data = cls.fetch_input_url()
+        # Make GET request and store the response in response_data.json
+        try:
+            response = requests.get(request_data["request_url"], headers= request_data["request_headers"])
+            response_data = cls.print_response_json(response)
+            cls.save_response_data(response_data)
+
+        except requests.exceptions.InvalidSchema:
+            print(cls.InvalidSchemaMessage)
+        except Exception as e:
+            print(e)
+    # Make a delete request
+    @classmethod
+    def delete_endpoint_request(cls):
+        # request_data contains dictionary of inputs entered by user
+        request_data = cls.fetch_input_url()
+        try:
+            response = requests.delete(request_data["request_url"], headers= request_data["request_headers"])
+            response_data = cls.print_response_json(response)
+            cls.save_response_data(response_data)
 
         except requests.exceptions.InvalidSchema:
-            print("Check whether the URL is valid or check if " +
-                  "the localhost server is active or not")
+            print(cls.InvalidSchemaMessage)
         except Exception as e:
```

Comment:
Can we change this 'e' as well to something meaningful ?

Reply:
yeah, better to name it ```unknown_error```

Reply:
Name it as exception_obj